modules/compatibility/src/test/java/org/apache/ignite/compatibility/persistence/SnapshotCompatibilityTest.java

```diff
@@ -0,0 +1,459 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.compatibility.persistence;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.annotation.Nullable;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.binary.BinaryType;
+import org.apache.ignite.cdc.TypeMapping;
+import org.apache.ignite.cluster.ClusterState;
+import org.apache.ignite.compatibility.testframework.junits.IgniteCompatibilityAbstractTest;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.DataStorageConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.dump.DumpConsumer;
+import org.apache.ignite.dump.DumpEntry;
+import org.apache.ignite.dump.DumpReader;
+import org.apache.ignite.dump.DumpReaderConfiguration;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.processors.cache.StoredCacheData;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.lang.IgniteInClosure;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+/**
+ *
+ */
+@RunWith(Parameterized.class)
+public class SnapshotCompatibilityTest extends IgniteCompatibilityAbstractTest {
+    /** */
+    @Parameterized.Parameter
+    public boolean incrementalSnp;
+
+    /** */
+    @Parameterized.Parameter(1)
+    @Nullable public String consistentId;
+
+    /** */
+    @Parameterized.Parameter(2)
+    public int oldNodesCnt;
+
+    /** */
+    @Parameterized.Parameter(3)
+    public boolean cacheDump;
+
+    /** */
+    @Parameterized.Parameter(4)
+    public boolean customSnpPath;
+
+    /** */
+    @Parameterized.Parameter(5)
+    public boolean testCacheGrp;
+
+    /** */
+    private static final String OLD_IGNITE_VERSION = "2.16.0";
+
+    /** */
+    private static final String SNAPSHOT_NAME = "test_snapshot";
+
+    /** */
+    private static final String CACHE_DUMP_NAME = "test_cache_dump";
+
+    /** */
+    private static final int BASE_CACHE_SIZE = 10_000;
+
+    /** */
+    private static final int ENTRIES_CNT_FOR_INCREMENT = 10_000;
+
+    /** */
+    private static final String CUSTOM_SNP_RELATIVE_PATH = "ex_snapshots";
+
+    /** */
+    private static final String CONSISTENT_ID = "db3e5e20-91c1-4b2d-95c9-f7e5f7a0b8d3";
```

Comment:
Let's replace this with the `private static final String CONSISTENT_ID = UUID.randomUUID().toString();`